cardano-cli/app/cardano-cli.hs

```diff
@@ -9,13 +14,17 @@ import           Cardano.CLI.Parsers (opts, pref)
 import           Cardano.CLI.Run (renderClientCommandError, runClientCommand)
 import           Cardano.CLI.TopHandler
 import qualified Cardano.Crypto.Libsodium as Crypto
-
+#ifdef UNIX
+import           System.Posix.Files
+#endif
 
 main :: IO ()
 main = toplevelExceptionHandler $ do
   -- TODO: Remove sodiumInit: https://github.com/input-output-hk/cardano-base/issues/175
   Crypto.sodiumInit
-
+#ifdef UNIX
+  _ <- setFileCreationMask (otherModes `unionFileModes` groupModes)
+#endif
```

Comment:
What do we need this for? The `openTempFile` makes files with the right permissions irrespective of the default file creation mask doesn't it?

Reply:
When I run a script that uses `cardano-cli`, the file appears to be created with the incorrect permissions. I either have to put `umask 077` at the top of my script or use `setFileCreationMask` in the `cardano-cli` exec.

Reply:
I cannot reproduce this.

Irrespective of the initial umask, the file is created without "other" or "group" getting any permissions. The only time I can get the umask to affect things is if I set it to a nonsensical value  like 0777 which prevents even the owner from having any permissions. But this is not a case we need to deal with.

That said, irrespective of whether it's needed or not, it may not be a bad idea to override the umask anyway and use 0077 for all files the cli creates, since arguably all the files the cli makes should be readable only to the user creating them.

@disassembler any opinion?